### ISSUE TITLE
第三章的完整神经网络样例程序结果与预期不符

### ISSUE THREAD

**Issue by yedf** [caicloud/tensorflow-tutorial#38]
在代码中，模型训练结束之后，进行验证时，结果与预期相差很大：
print(sess.run(y, feed_dict={x: [[0.8, 0.8], [0.1, 0.1], [0.4, 0.4], [0.5, 0.5]]}))
[[ 20.08403206]
 [  2.51050401]
 [ 10.04201603]
 [ 12.55251884]]
与训练集中期望的x1+x2<1的0/1不同
建议的patch如下
：
 w2 = tf.Variable(tf.random_normal([3, 1], stddev=1, seed=1))
+b = tf.Variable(tf.random_normal([1, 1], stddev=1, seed=1))
 
 x = tf.placeholder(tf.float32, (None, 2), 'x-input')
 y_ = tf.placeholder(tf.float32, (None, 1), 'y-input')
 
 a = tf.matmul(x, w1)
 y = tf.matmul(a, w2)
+y = tf.sigmoid(y + b)
 
-cross_entropy = -tf.reduce_mean(y_ * tf.log(tf.clip_by_value(y, 1e-10, 1.0)))
+cross_entropy = tf.reduce_mean(tf.square(y_ - y))
 train_step = tf.train.AdamOptimizer(0.001).minimize(cross_entropy)

**Comment by perhapszzy** [comment]
这里加上一个bias、激活函数并且将损失函数改成MSE确实会使得结果更好，欢迎提交一个pr将样例代码进行修改。

**Comment by ScorpioCPH** [comment]
@yedf ping.

**Comment by Hailprob** [comment]
问一下程序中设置每个batch的start为什么用

>  start = (i*batch_size)%dataset_size
而不用

> start = (i*batch_size)
巧妙之处在何处？
